modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/RocksIteratorAdapter.java

```diff
@@ -53,7 +55,11 @@ public boolean hasNext() {
         if (!isValid) {
             // check the status first. This operation is guaranteed to throw if an internal error has occurred during
             // the iteration. Otherwise, we've exhausted the data range.
-            RocksUtils.checkIterator(it);
+            try {
+                it.status();
+            } catch (RocksDBException e) {
+                throw new IgniteInternalException(e);
```

Comment:
I found that this iterator is redefined by different modules, maybe it is bad to throw a common error, what do you think?
Maybe for example for the metastore iterator it is necessary to throw its error?

Reply:
I also don't like it, but `RocksDBException` is a checked exception, so we can't throw it directly here (this method is from an interface)